### DISCUSSION TITLE
Urgent: Scheduled task runs multiple instances when timer triggered

### DISCUSSION THREAD

**Discussion by younky-yang** [directus/directus#17674] [Archive]
Hi, All, 
I created a schedule hook for my directus instance which sending emails to some users when something happen. The challenge is the users will receive 2 emails with the same content. After investigation I found the reason is we have multiple instance of directus container running on AWS, and every time the schedule triggered, each instance will send email to the users, so for my testing environment, the user receives 2 identical emails. 

So how I can control the schedule hook only runs on one instance, and still allow me to have multiple instance deployment?

**Comment by domdus** [comment]
https://github.com/directus/directus/issues/15052#issuecomment-1213307284

**Reply by younky-yang** [reply]
Thanks, seems I need to solve this by my own schedule hook. Will try to add lock to my db table.
